app/src/main/java/org/mozilla/focus/tiles/HomeTilesManager.kt

```diff
@@ -7,13 +7,90 @@ package org.mozilla.focus.tiles
 import android.content.Context
 import android.content.Context.MODE_PRIVATE
 import android.content.SharedPreferences
+import android.graphics.BitmapFactory
 import android.support.annotation.UiThread
 import org.json.JSONArray
+import org.json.JSONObject
+import org.mozilla.focus.utils.UrlUtils
+import java.net.URL
 import java.util.UUID
 
 private const val PREF_HOME_TILES = "homeTiles"
 private const val CUSTOM_SITES_LIST = "customSitesList"
 
+private const val BUNDLED_HOME_TILES_DIR = "bundled/"
+private const val HOME_TILES_JSON_PATH = BUNDLED_HOME_TILES_DIR + "bundled_tiles.json"
+private const val HOME_TILES_JSON_KEY = "bundled_tiles"
+
+class BundledTilesManager private constructor(context: Context) {
+    companion object {
+        private var thisInstance: BundledTilesManager? = null
+        fun getInstance(context: Context): BundledTilesManager {
+            if (thisInstance == null) {
+                thisInstance = BundledTilesManager(context)
+            }
+            return thisInstance!!
+        }
+    }
+
+    private var bundledTilesCache = loadBundledTilesCache(context)
+
+    private fun loadBundledTilesCache(context: Context): LinkedHashMap<URL, BundledHomeTile> {
```

Comment:
We don't use the HashMap part of this at all - we just iterate over it. Should we make it a list? This works for now so maybe it's not worth it...

Reply:
We need the whole bundled tile for 1) passing back to the home screen every time we load it, and 2) (future) for deleting tiles by id.

Reply:
It can be a [BundledHomeTile] though since we only ever iterate and don't call `contains` or `get` (except after iterating to find the key), afaict.